[AIRFLOW-3765] Fix for create/update xcom values using the admin UI

andyhuynh3 opened the pull request:
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3765
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

Refactoring some of the serialization / deserialization logic into a custom SQLAlchemy TypeDecorator so that creating / updating XCom values using the Admin UI works again.

Creating new XCom entry:
![image](https://user-images.githubusercontent.com/8610246/51798248-a52ce400-21c4-11e9-88ff-e2df20498a89.png)

Updating XCom entry:
![image](https://user-images.githubusercontent.com/8610246/51798266-f9d05f00-21c4-11e9-8a0a-43741b1ab851.png)


### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason: There already exists XCom tests.

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [x] Passes `flake8`

Commit:
[AIRFLOW-3771] Minor refactor securityManager (#4594)

Commit:
[AIRFLOW-XXX] Remove images related profiling doc (#4599)

andyhuynh3 committed:
Fix for create/update xcom values using the admin UI

andyhuynh3 committed:
Fix for create/update xcom values using the admin UI

XD-DENG committed:
[AIRFLOW-3761] Decommission User & Chart models & Update doc accordingly (#4577)

In master branch, we have already decommissioned the Flask-Admin UI.

In model definitions, User and Chart are only applicable for the
"old" UI based on Flask-Admin.
Hence we should decommission these two models as well.

Related doc are updated in this commit as well.

mik-laj committed:
[AIRFLOW-XXX] Remove almost all warnings from building docs (#4588)

mik-laj committed:
[AIRFLOW-XXX] Remove profiling link (#4602)

astahlman committed:
[AIRFLOW-3773] Fix /refresh_all endpoint (#4597)

* [AIRFLOW-3773] Fix /refresh_all endpoint

Call `sync_perm_for_dag` for each DAG in the DagBag (`dag_id` is a
required argument).

I looked for a test suite for the web UI, but it seems the existing
tests have all been disabled since the switch to FAB. I've created a new
class for FAB tests and added a test to exercise this `/refresh_all`
endpoint.

* Move tests to www/test_views.py

I didn't realize that we already had test scaffolding in place for
testing the FAB-based UI.

seelmann commented on airflow/models/__init__.py:
```diff
@@ -188,6 +188,55 @@ def get_fernet():
     return _fernet
 
 
+class XComSerializer(types.TypeDecorator):
```
I'd suggest to directly move this new class out into it's own file?
See discussion https://lists.apache.org/thread.html/3c15a871b15f99b4dd4bef25085b4de1c167c728f08262df73ef7217@%3Cdev.airflow.apache.org%3E

andyhuynh3 replied:
Hey @seelmann, thanks for the suggestion. That's a good idea. Wanted to check if it also makes sense to move the XCom model / class to this file as well (i.e. airflow/models/xcom.py would contain the XCom class and XComSerializer class).

r39132 committed:
[AIRFLOW-XXX] Add Tinder to the companies list (#4604)

anoopengineer committed:
[AIRFLOW-XXX] Add Capital One to the companies list (#4606)

ryanyuan committed:
[AIRFLOW-3762] Add list_jobs to CLI (#4579)

* [AIRFLOW-3762] Add list_jobs to CLI

Add list_jobs to CLI

* [AIRFLOW-3762] Add list_jobs to CLI

Improve test_cli_list_jobs_with_args

*  [AIRFLOW-3762] Add list_jobs to CLI

Directly parse args.limit to list_jobs query

* [AIRFLOW-3762] Add list_jobs to CLI

Format list_jobs code

andyhuynh3 committed:
[AIRFLOW-3765] Fix merge conflicts; Move custom SQLAlchemy types to its own file
